### PR TITLE
Add instructions for upgrading when installed via git clone

### LINKED ISSUE #250: [upgrading.xml] Add upgrading instructions for installations from source.
It would be useful to add instructions about how to update/upgrade eXist-db for people who installed eXist-db via `git clone git://github.com/eXist-db/exist.git`.

Add a section titled something like _Installing and Upgrading from Source_ with the following instructions.

> In the folder where you installed eXist-db, run `git status` to see which files differ from the updated repository on GitHub. In particular, you're interested to see which are the configuration files you edited and which you want to preserve (e.g. `conf.xml`, `webapp/WEB-INF/controller-config.xml`, `tools/jetty/webapps/exist-webapp-context.xml`, etc.).
> 
> 1. Create a full backup from eXist-db (i.e. to a set of files or .zip file).
> 2. Stop eXist-db
> 3. Make a backup copy of the `webapp/WEB-INF/data directory` directory and any other configuration file you wish to preserve (see above), plus the 'local.build.properties' file in case you edited `build.properties`.
> 4. Run `git checkout -- lib/` to unsign the jar files.
> 5. Run `git status`: you should see no lib/ paths in the output.
> 6. Run `git fetch`.
> 7. Run `git stash`.
> 8. Run `git rebase origin/master` or `develop`, whichever you want.
> 9. Run `git stash pop`.
> 10. Run `./build.sh clean`.
> 11. Run `./build.sh jnlp-unsign-all all jnlp-sign-exist jnlp-sign-core jnlp-sign-exist-extensions`.
> 12. Copy back the configuration files (and, optionally, the' local.build.properties` - see above).
> 12. Start your eXist-db and you should now have an up-to-date eXist-db instance running.

### PR DESCRIPTION
**Pull request by lguariento**
Closes eXist-db/documentation/issues/#250

### CODE REVIEW

**Comment by duncdrum** [comment]
Hi Luca thanks for the updated PR. I ve attached some comments in my review. It would be very helpful if you could add a few sentences about who this is for and when it might matter. A list of command line arguments is rarely useful as documentation for beginners.

**Comment by lguariento** [comment]
Hi @duncdrum , thanks for your feedback. Since I'm not sure myself what each single command does, I was hoping you guys could add some extras.

As for who this is and why one needs it, since it is part of the 'upgrading' guide I assumed it was already clear at the beginning of that page. Am I wrong? Any suggestion welcome.

**Comment by duncdrum** [comment]
@lguariento understood we'll take it from here, thanks again. The other sections of the document apply to every upgrade scenario, this section only applies to folks building and running from source, so we need to make that clear.

File: src/main/xar-resources/data/upgrading/upgrading.xml [hunk 1]
```diff
@@ -1,209 +1,234 @@
...
-      version (as can be found in the release notes). The key difference with upgrading to a binary-compatible new version
-      is that you have to perform a <emphasis>full restore</emphasis> of your backed-up data into the
-      new installation.</para>
-    <orderedlist>
-      <listitem>
-        <para>Create a <link xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="backup">server-side backup</link>. This can be done in a number
-          of ways: </para>
+    <info>
+        <title>Upgrade Guide</title>
+        <date>1Q18</date>
```
Comment (duncdrum) [inline src/main/xar-resources/data/upgrading/upgrading.xml]: needs update

File: src/main/xar-resources/data/upgrading/upgrading.xml [hunk 2]
```diff
@@ -1,209 +1,234 @@
...
-      foir instance from 1.4 to 3.6, we strongly recommend to first check past release notes to understand
-      how breaking changes might affect you.</para>
+    </sect1>
+    
+    <!-- ================================================================== -->
+    
+    <sect1 xml:id="upgrading-from-source">
+        <title>Upgrading from Source</title>
+        <orderedlist>
+
```
Comment (duncdrum) [inline src/main/xar-resources/data/upgrading/upgrading.xml]: needs an intro, to make clear that it only applies to folks building and running from source

File: src/main/xar-resources/data/upgrading/upgrading.xml [hunk 3]
```diff
@@ -1,209 +1,234 @@
...
-
-    <para>This is a list of binary non-compatible releases for quick reference.</para>
-    <para>If a non-incremental upgrade across multiple non-compatible releases is necessary,
-      foir instance from 1.4 to 3.6, we strongly recommend to first check past release notes to understand
-      how breaking changes might affect you.</para>
+    </sect1>
+    
+    <!-- ================================================================== -->
+    
+    <sect1 xml:id="upgrading-from-source">
```
Comment (duncdrum) [inline src/main/xar-resources/data/upgrading/upgrading.xml]: should probably become lvl2

File: src/main/xar-resources/data/upgrading/upgrading.xml [hunk 4]
```diff
@@ -1,209 +1,234 @@
...
+        <title>Upgrading from Source</title>
+        <orderedlist>
+            
+            <listitem><para>In the folder where you installed eXist-db, run <code>git status</code> to see which files differ from the updated repository on GitHub. In particular, you're interested to see which are the configuration files you edited and which you want to preserve (e.g. <code>conf.xml</code>, <code>webapp/WEB-INF/controller-config.xml</code>, <code>tools/jetty/webapps/exist-webapp-context.xml</code>, etc.).</para></listitem>
```
Comment (duncdrum) [inline src/main/xar-resources/data/upgrading/upgrading.xml]: We should try to trim the list where possible

**Comment by lguariento** [comment]
I'll try to re-word my thought: is _upgrade for installations from source_ a subsection of the higher_upgrading instructions for eXist-db_, or of _non binary-compatible upgrades_, of _binary-compatible upgrades_, or none of the above? Or all of them?

**Comment by duncdrum** [comment]
superseded by #254

**Pull request closed without merging**